src/API/Functions/Units.cs

```diff
@@ -38,8 +40,64 @@ public static class Units
         public static Task<IActionResult> UnitsGetOne(
             [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "units/{unitId}")] HttpRequest req, int unitId) 
             => Security.Authenticate(req)
-                .Bind(requestor => AuthorizationRepository.DetermineUnitPermissions(req, requestor, unitId))
+                .Bind(requestor => AuthorizationRepository.DetermineUnitPermissions(req, requestor, unitId))// Set headers saying what the requestor can do to this unit
                 .Bind(_ => UnitsRepository.GetOne(unitId))
                 .Finally(result => Response.Ok(req, result));
+        
+        [FunctionName(nameof(Units.CreateUnit))]
+        [OpenApiOperation(nameof(Units.CreateUnit), nameof(Units), Summary = "Create a unit", Description = "_Authorization_: Unit creation is restricted to service administrators.")]
+        [OpenApiParameter("name", Type = typeof(string), In = ParameterLocation.Query, Required = true, Description = "")]
+        [OpenApiParameter("description", Type = typeof(string), In = ParameterLocation.Query, Required = false, Description = "")]
+        [OpenApiParameter("url", Type = typeof(string), In = ParameterLocation.Query, Required = false, Description = "")]
+        [OpenApiParameter("email", Type = typeof(string), In = ParameterLocation.Query, Required = false, Description = "")]
+        [OpenApiParameter("parentId", Type = typeof(int), In = ParameterLocation.Query, Required = false, Description = "The Unit Id of the parent Unit.")]
```

Comment:
We can use an `OpenApiRequestBody` attribute to document the shape of the request body. I like the use of these annotations to document the properties since that's obviously lacking in the OpenAPI spec generator. Unfortunately the parameters aren't in the query string as indicated here, which might confuse some clients.

Reply:
A picture's worth a thousand words.  Mind sending up a commit with it fixed?

Reply:
Yep, will do